util/endpointParams.js

```diff
@@ -0,0 +1,34 @@
+const { chainTableNames } = require('./constants.js');
+const { dbQueryAvgTxCost } = require('./activity-report/index.js');
+
+function verifyChain(chain) {
+  if (!chain || !chainTableNames[chain]) {
+    throw new Error(`chain '${chain}' is not supported`);
+  }
+}
+
+module.exports = {
+  'api/v1/rsk-activity-report/avg-tx-cost': {
+    cacheTtl: 600, // seconds
+    queryDb: dbQueryAvgTxCost,
+    queryStringParams: [
+      {
+        name: 'chain',
+        verify: verifyChain,
+      },
+      {
+        name: 'blocks',
+        verify: (blocks) => {
+          const blocksRange = {
+            lower: 1,
+            upper: 1000,
+          };
+          if (!(blocks >= blocksRange.lower && blocks <= blocksRange.upper))
+            throw new Error(
+              `Number of blocks should be within range ${blocksRange.lower} to ${blocksRange.upper}, specified value was: ${blocks}`,
+            );
+        },
```

Comment:
I think the cache middleware only needs to be aware of the names of the query params that are significant. all other logic (including verification) is specific to the individual APIs, and not relevant to caching.
probably better to keep that within the individual API impls rather than in this config file.

Reply:
does it make sense why QS params verification needs to be done before sending a response with cached values? Or you can suggest some other approach?

Reply:
I think that the approach that you've done in your latest commit makes sense, because the logic for verification has been moved to the API implementations (which is what should define the valid input values for their params)